### DISCUSSION TITLE
How to get local files for speechbrain model artifacts

### DISCUSSION THREAD

**Discussion by JamesBowerXanda** [speechbrain/speechbrain#2546] [Q&A]
Hi,

I am trying to package up a speechbrain model with some code so that it can be moved and loaded locally on a sagemaker instance. The issue I am having is that the file in the savdir folder seem to be links rather than the actual files themselves.

This means that packaging them up and moving to another environment is useless. Does anyone know how to get speechbrain to download the relevant model file s into the current working directory so that I can package them up to be loaded at runtime in a different environment?

The model in question is speechbrain/spkrec-ecapa-voxceleb.

James

**Comment by asumagic** [comment]
Currently, there is no built-in functionality for this in SpeechBrain, which will always symlink in the save directory. #2476 will help address this usecase, but it is not ready/merged yet.

You could just use the `cp` command with [the proper parameters](https://superuser.com/questions/216919/how-to-copy-symlinks-to-target-as-normal-folders) to copy your savedir somewhere else while also resolving symlinks (which point to files in the HuggingFace Hub cache).

Alternatively, since HF repos are just git repositories, you can do something like `git clone https://huggingface.co/speechbrain/spkrec-ecapa-voxceleb` (note that you need git LFS installed).

**Comment by JamesBowerXanda** [comment]
Ok, thanks for getting back to me.
